python/ray/dag/dag_node.py

```diff
@@ -335,6 +369,39 @@ def __call__(self, node: "DAGNode"):
             )
         )
 
+    def bfs(self, fn: "Callable[[DAGNode], T]"):
+        """Breadth-first search on the DAG."""
```

Comment:
nit: Breadth-first search traversal of the DAG, and apply `fn` to all nodes.

Reply:
update: https://github.com/ray-project/ray/pull/47261/commits/cfeb6020eb5951916ce4ee9a8db9a20a1cdd04ae